src/net_processing.cpp

```diff
@@ -3122,10 +3122,11 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
             // Determine transactions to relay
             if (fSendTrickle) {
                 // Produce a vector with all candidates for sending
-                std::vector<std::set<uint256>::iterator> vInvTx;
+                std::vector<uint256> vInvTx;
                 vInvTx.reserve(pto->setInventoryTxToSend.size());
-                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); it != pto->setInventoryTxToSend.end(); it++) {
-                    vInvTx.push_back(it);
+                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); it != pto->setInventoryTxToSend.end(); ) {
+                    vInvTx.push_back(*it);
+                    it = pto->setInventoryTxToSend.erase(it);
```

Comment:
I suppose things are broken after commit "[refactor] Use uint256 vector instead of set iterator vector in trick…" but fixed later? Looks like it'll wipe setInventoryTxToSend always, whereas previously it oulw only wipe the first INVENTORY_BROADCAST_MAX entries and retry the rest later, and then the next commit fixes it by limiting this loop to INVENTORY_BROADCAST_MAX.